client/state/notices/middleware.js

```diff
@@ -98,6 +157,14 @@ export const handlers = {
 	[ POST_RESTORE_FAILURE ]: onPostRestoreFailure,
 	[ POST_RESTORE_SUCCESS ]: dispatchSuccess( translate( 'Post successfully restored' ) ),
 	[ POST_SAVE_SUCCESS ]: onPostSaveSuccess,
+	[ PUBLICIZE_CONNECTION_CREATE ]: onPublicizeConnectionCreate,
+	[ PUBLICIZE_CONNECTION_CREATE_FAILURE ]: dispatchErrorMessage,
```

Comment:
Looking at the endpoint implementation, the messages would be translated if we provided a locale parameter (perhaps via [`wpcom.withLocale`](https://github.com/Automattic/wp-calypso/tree/master/client/lib/wp/localization)). I agree with @gwwar though that the messages aren't particularly user friendly, e.g.

>Keyring connection with this ID not found.